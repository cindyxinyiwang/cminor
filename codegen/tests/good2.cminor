// scope
i:integer = 1;
j:integer = i+1;
k:boolean;
a: function integer (i: integer, j:boolean) = {
	return i*2;

}
b: function void () = {
	x:integer = 1;
	i:boolean = true;
	j:boolean = i;
	k = i;
	m: array [4] integer = {1, 2, 3, 4};
	x = a(x, j);
	{
		i: string = "hello";
		j = i=="hello";
	}
}