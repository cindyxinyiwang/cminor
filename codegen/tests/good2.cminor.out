i resovles to global i
i resovles to param 0
i resovles to local 1
k resovles to global k
i resovles to local 1
x resovles to local 0
x resovles to local 0
j resovles to local 2
a resovles to global a
j resovles to local 2
i resovles to local 4
